src/main/flight/autopilot.c

```diff
@@ -22,38 +22,149 @@
-    altitudePidCoeffs.Ki = config->altitude_I * ALTITUDE_I_SCALE;
-    altitudePidCoeffs.Kd = config->altitude_D * ALTITUDE_D_SCALE;
-    altitudePidCoeffs.Kf = config->altitude_F * ALTITUDE_F_SCALE;
+    // at start only
+    resetEFAxisFilters(efAxis, vaGain);
+    efAxis->isStopping = true; // Enter starting (deceleration) 'phase'
+    efAxis->integral = 0.0f;
+}
+
+static void resetUpsampleFilters(void)
+{
+    for (unsigned i = 0; i < ARRAYLEN(ap.upsampleLpfBF); i++) {
+        pt3FilterInit(&ap.upsampleLpfBF[i], ap.upsampleLpfGain);
+    }
+}
+
+// get sanity distance based on speed
+static inline float sanityCheckDistance(const float gpsGroundSpeedCmS)
+{
+    return fmaxf(1000.0f, gpsGroundSpeedCmS * 2.0f);
+    // distance flown in 2s at current speed. with minimum of 10m
+}
+
+void resetPositionControl(const gpsLocation_t *initialTargetLocation, unsigned taskRateHz)
+{
+    // from pos_hold.c (or other client) when initiating position hold at target location
+    ap.targetLocation = *initialTargetLocation;
+    ap.sticksActive = false;
+    // set sanity check distance according to groundspeed at start, minimum of 10m
+    ap.sanityCheckDistance = sanityCheckDistance(gpsSol.groundSpeed);
+    for (unsigned i = 0; i < ARRAYLEN(ap.efAxis); i++) {
+        // clear anything stored in the filter at first call
+        resetEFAxisParams(&ap.efAxis[i], 1.0f);
+    }
+    const float taskInterval = 1.0f / taskRateHz;
+    ap.upsampleLpfGain = pt3FilterGain(UPSAMPLING_CUTOFF_HZ, taskInterval); // 5Hz; normally at 100Hz task rate
+    resetUpsampleFilters(); // clear accumlator from previous iterations
+}
+
+void autopilotInit(void)
+{
+    const apConfig_t *cfg = apConfig();
+
+    ap.sticksActive = false;
+    ap.maxAngle = cfg->max_angle;
+    altitudePidCoeffs.Kp = cfg->altitude_P * ALTITUDE_P_SCALE;
+    altitudePidCoeffs.Ki = cfg->altitude_I * ALTITUDE_I_SCALE;
+    altitudePidCoeffs.Kd = cfg->altitude_D * ALTITUDE_D_SCALE;
+    altitudePidCoeffs.Kf = cfg->altitude_F * ALTITUDE_F_SCALE;
+    positionPidCoeffs.Kp = cfg->position_P * POSITION_P_SCALE;
+    positionPidCoeffs.Ki = cfg->position_I * POSITION_I_SCALE;
+    positionPidCoeffs.Kd = cfg->position_D * POSITION_D_SCALE;
+    positionPidCoeffs.Kf = cfg->position_A * POSITION_A_SCALE; // Kf used for acceleration
+    // initialise filters with approximate filter gains; location isn't used at this point.
+    ap.upsampleLpfGain = pt3FilterGain(UPSAMPLING_CUTOFF_HZ, 0.01f); // 5Hz, assuming 100Hz task rate at init
+    resetUpsampleFilters();
+    // Initialise PT1 filters for deivative and acceleration in earth frame axes
+    ap.vaLpfCutoff = cfg->position_cutoff * 0.01f;
+    const float vaGain = pt1FilterGain(ap.vaLpfCutoff,  0.1f); // assume 10Hz GPS connection at start; value is overwritten before first filter use
+    for (unsigned i = 0; i < ARRAYLEN(ap.efAxis); i++) {
+        resetEFAxisFilters(&ap.efAxis[i], vaGain);
+    }
```

Comment:
I already tried that, in b51ae1395d  `use a null location at initialisation`, with these lines:
```
    ap.vaLpfCutoff = cfg->position_cutoff * 0.01f;
    const gpsLocation_t initialiseLocation = {0};  // somewhere off the coast of Guinea
    resetPositionControl(&initialiseLocation, 100);
```
It also failed the SITL test.  I tried again with your exact suggestion in case my comment about Guinea was imperialist.  Same result.